python/pip_install/pip_repository.bzl

```diff
@@ -416,6 +416,56 @@ def _pip_repository_impl(rctx):
 
     return
 
+def _try_finding_platform_agnostic_wheel(rctx, requirement):
+    """Tries to find a platform agnostic wheel for the given requirement, otherwise returns None"""
+    split_req = [i for i in requirement.split(" ") if i != ""]
+    requirement_name, requirement_version = split_req[0].split("==")
+    if "[" in requirement_name:
+        # Remove 'extras' specification
+        requirement_name = requirement_name.split("[")[0]
+
+    # Get all possible expected hashes
+    requirement_hashes = [s.removeprefix("--hash=") for s in split_req[1:]]
+    split_hashes = [s.split(":") for s in requirement_hashes]
+    by_hash_type = {}
+    for alg, digest in split_hashes:
+        if alg in by_hash_type:
+            by_hash_type[alg].append(digest)
+        else:
+            by_hash_type[alg] = [digest]
+
+    rctx.download(
+        url = "https://pypi.org/pypi/{}/{}/json".format(requirement_name, requirement_version),
```

Comment:
There are many people that use private or alternative indexes; hard-coding `pypi.org` wouldn't work in these scenarios.

Also, I'm not sure that the JSON api is widely supported outside of pypi. [Artifactory](https://jfrog.com/artifactory/) is a common repository and it only supports the simple (HTML) API as far as I can tell.

Reply:
I was using the simple API  and switched to using the JSON API because it was far simpler to just parse the output for one specific version :shrug:  I wasn't really aware it was widely unsupported